app/uk/gov/hmrc/gform/sharedmodel/form/Form.scala

```diff
@@ -51,8 +53,15 @@ object Form {
   val readVisitIndex: Reads[VisitIndex] =
     (__ \ "visitsIndex").readNullable[List[Int]].map(a => VisitIndex(a.fold(Set.empty[Int])(_.toSet)))
 
+  val readerRetreived: Reads[Option[Obligations]] = (__ \ "RetrievedObligations" \ "listOfObligations")
+    .readNullable[List[TaxPeriodInformation]]
+    .map(_.map(RetrievedObligations))
+  val readerStandard: Reads[Option[Obligations]] =
+    (__ \ "obligations").readNullable[List[TaxPeriodInformation]].map(_.map(RetrievedObligations))
```

Comment:
There is something fishy going on. Why we read `Obligations` at one place as `"RetrievedObligations" \ "listOfObligations"` and at second place `"obligations"`?

Reply:
this is what I was asking you about before lunch, if we do not have the reads \ "obligations" the frontend gets an error page thrown..

Reply:
I imagine this would be something related with how the data is stored and then attempted to retrieve.

Reply:
Yes, of course, data in master are using `"obligations"` as a key in json. We need to write unit test to cover if we are supporting all possible states.

1. Having no obligation in mongo.
2. Having obligations under `"obligations"` key (as result of using `Json.format` macro).
3. Having obligations under `"RetrievedObligations" \ "listOfObligations"` key (as result of using  `derived.oformat` macro)